test/integration/scheduler/predicates_test.go

```diff
@@ -39,7 +42,8 @@ const pollInterval = 100 * time.Millisecond
 // TestInterPodAffinity verifies that scheduler's inter pod affinity and
 // anti-affinity predicate functions works correctly.
 func TestInterPodAffinity(t *testing.T) {
-	testCtx := initTest(t, "inter-pod-affinity")
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.PodAffinityNamespaceSelector, true)()
```

Comment:
test the feature with the featuregate turned off? I think integration tests are the only place we can test both

Reply:
We have coverage under unit tests. The issue with testing both modes is that each test case would need to instantiate and start the scheduler and api-server from scratch, and that has significant impact on the runtime of the test. May be we should split the namespaceselector cases into their own test?

Reply:
I split them into their own test, take a look at the last commit.

Reply:
hmm, that creates a lot of code repetition, but I don't see any way around it if we want to keep test runtime small. lgtm, thanks (will do another full pass)